snapd-glib/snapd-client-private.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2025 Canonical Ltd.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2 or version 3 of the License.
+ * See http://www.gnu.org/copyleft/lgpl.html the full text of the license.
+ */
+
+#pragma once
+
+/*
+ * This function is declared in this private header because it must be private,
+ * but also is needed in the tests.
+ */
+
+static GSocket *open_snapd_socket(const gchar *socket_path,
```

Comment:
There's no need to put the whole function in the header, just remove the static from the .c file and put the function prototype here. The function won't be made public in the library as it doesn't start with `snapd_` (see `snapd-glib/snapd-glib.map`.

However, I would just copy this code into the tests. There's no real reason to make it shared.

Reply:
That's what I tried before, but didn't work because the tests seem to link with the shared library instead of using the object files (during compilation it didn't find the function). That's why first I added an extra function that was published only when building the tests and putting the declaration directly in the test code, and then followed the idea from Marco of just putting the code in a private header.

About copying this code into the tests, I disagree, because any change done in the library would require remembering to do the same change in the tests, due to the code duplication.

Reply:
Looking at the test in more detail shows why - it doesn't make sense to be testing an internal part of the SnapdClient logic. The tests should modify MockSnapd to support an abstract socket and then do something simple like `snapd_client_get_system_information_sync` and confirm it works.